File: code/generate_strategies/Direct/python/function_Polars_filter.py
import polars as pl


class PolarsExample:
    def filter_numbers(self, threshold: float, x: float, y: float = None, z: float = None) -> float:
        values = [val for val in [x, y, z] if val is not None]

        if not values:
            return None

        df = pl.DataFrame({"values": values})

        filtered_df = df.filter(pl.col("values") > threshold)

        if filtered_df.height == 0:
            return None

        return filtered_df.select(pl.col("values").max()).to_numpy()[0][0]

File: code/generate_strategies/RA(name)/java/java/function_Polars_filter.java
package com.example;

import org.apache.commons.math3.stat.descriptive.rank.Max;
import org.apache.commons.math3.util.MathArrays;
import java.util.ArrayList;
import java.util.List;

public class PolarsExample {
    public Float filter_numbers(float threshold, float x, Float y, Float z) {
        // 收集非null的值
        List<Float> valuesList = new ArrayList<>();
        valuesList.add(x);
        if (y != null) valuesList.add(y);
        if (z != null) valuesList.add(z);

        if (valuesList.isEmpty()) {
            return null;
        }

        // 转换为double数组并过滤
        double[] values = valuesList.stream()
                .mapToDouble(Float::doubleValue)
                .filter(val -> val > threshold)
                .toArray();

        if (values.length == 0) {
            return null;
        }

        // 使用Apache Commons Math来找最大值
        Max maxCalculator = new Max();
        double maxValue = maxCalculator.evaluate(values);

        return (float) maxValue;
    }
}